nuxeo-features/nuxeo-elasticsearch/nuxeo-elasticsearch-core/src/main/java/org/nuxeo/elasticsearch/ElasticSearchFilter.java

```diff
@@ -50,7 +50,9 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
             return;
         }
         final HttpServletRequest httpRequest = (HttpServletRequest) request;
-        final boolean esSync = Boolean.parseBoolean(httpRequest.getHeader(ES_SYNC_FLAG));
+        final boolean esSync = Boolean.parseBoolean(httpRequest.getHeader(ES_SYNC_FLAG))
+                // NXP-28075: keep compatibility with old header with underscores
+                || Boolean.parseBoolean(httpRequest.getHeader(ES_SYNC_FLAG.replace("-", "_")));
```

Comment:
I'd prefer a constant `ES_SYNC_FLAG_COMPAT = "nx_es_sync"` which also helps when we grep the code.

Reply:
thought about that.. will do 👍